velox/vector/VariantToVector.cpp

```diff
@@ -89,6 +111,12 @@ ArrayVectorPtr variantArrayToVector(
   }
 
   const auto elementType = variantArray.at(nonNullIndex.value()).inferType();
+  if (elementType->kind() == TypeKind::SHORT_DECIMAL) {
+    return variantArrayToVectorImpl<TypeKind::SHORT_DECIMAL>(
+        variantArray, pool);
+  } else if (elementType->kind() == TypeKind::LONG_DECIMAL) {
+    return variantArrayToVectorImpl<TypeKind::LONG_DECIMAL>(variantArray, pool);
+  }
```

Comment:
It's not possible to use `VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH_ALL`?

Reply:
No this hadn't worked before. I will rebase this on top of the merged changes by Masha and revert all current changes to this file.